File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/CosH.java
package uk.ac.uea.mathsthing.plugins.functions;

import java.math.BigDecimal;

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin;
import uk.ac.uea.mathsthing.util.FormulaException;

public class CosH extends IFunctionPlugin {
	
	@Override
	public String getName()
	{
		return "cosh";
	}

	@Override
	public BigDecimal function(BigDecimal input) throws FormulaException
	{
		return new BigDecimal(Math.cosh(input.doubleValue()));
	}

}

File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/CosH.scala
package uk.ac.uea.mathsthing.plugins.functions

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin
import java.math.BigDecimal

class CosH extends IFunctionPlugin {

  def getName : String = "cosh"
    
  def function(input: BigDecimal) : BigDecimal = {
  	val result:BigDecimal = new BigDecimal(Math.cosh(input.doubleValue))
    result
  }
}